qiskit/primitives/backend_estimator.py

```diff
@@ -186,30 +186,39 @@ def _transpile(self):
         self._transpiled_circuits = []
         for common_circuit, diff_circuits in self.preprocessed_circuits:
             # 1. transpile a common circuit
-            common_circuit = common_circuit.copy()
-            num_qubits = common_circuit.num_qubits
-            common_circuit.measure_all()
             if not self._skip_transpilation:
```

Comment:
It might be more readable swapping if caluses such as 
```python
            if self._skip_transpilation:
                transpiled_circuit = common_circuit.copy()
                perm_pattern = list(range(common_circuit.num_qubits))
            else:  # easier to know what `else` means because `if self._skip_transpilation` is just two lines above.
                ...
                if transpiled_circuit.layout is None:
                    perm_pattern = list(range(transpiled_circuit.num_qubits))
                else:
                    ...
```